crc32.go

```diff
@@ -16,6 +16,12 @@ func crc32OfMessage(magicByte int8, attributes int8, timestamp int64, key []byte
 	}
 	b.writeBytes(key)
 	b.writeBytes(value)
+
+	b.writeArray(len(headers), func(i int) {
+		b.writeBytes(headers[i].Key)
```

Comment:
I tried the option when the key is a string: https://kafka.apache.org/documentation/#record
But it also does not work